TopMovies/API/MovieAPI.swift

```diff
@@ -52,17 +54,29 @@ struct MovieAPI: MovieAPIProtocol {
   public func category(_ requestedCategory: MovieCategoryRequest,
                        page: Int,
                        _ category: @escaping (Result<CategoryDTO, Error>) -> Void) {
-    provider.request(.init(requestedCategory: .init(category: requestedCategory,
-                                                    page: page))) { (response) in
+    previewProvider.request(.init(requestedCategory: .init(category: requestedCategory,
+                                                           page: page))) { response in
       switch response {
       case let .success(data):
         if var moviesListDTO = DataDecoder.decode(CategoryDTO.self, fromJSON: data.data) {
           moviesListDTO.name = requestedCategory.rawValue
           category(.success(moviesListDTO))
-        } else { print("Decoder didn't decode") }
+        } else { print("Decoder didn't decode CategoryDTO") }
       case let .failure(error):
         category(.failure(error))
       }
     }
   }
+  public func movie(id: MoviePreview.ID, _ result: @escaping (Result<MovieDTO, Error>) -> Void) {
+    movieProvider.request(.init(movieId: id)) { response in
+      switch response {
+      case let .success(response):
+        if let movie = DataDecoder.decode(MovieDTO.self, fromJSON: response.data) {
+          result(.success(movie))
+        } else { print("Decoder didn't decode MovieDTO") }
```

Comment:
Avoid using all prints in your project. Literally - no prints )
Instead, use custom errors and pass them in state:

```swift
struct ParsingError: Error {}
result(.failure(ParsingError()))
```